source/common/http/conn_manager_impl.cc

```diff
@@ -1282,6 +1282,29 @@ void ConnectionManagerImpl::ActiveStream::refreshCachedRoute() {
         connection_manager_.cluster_manager_.get(stream_info_.route_entry_->clusterName());
     cached_cluster_info_ = (nullptr == local_cluster) ? nullptr : local_cluster->info();
   }
+  refreshCachedTracingCustomTags();
+}
+
+void ConnectionManagerImpl::ActiveStream::refreshCachedTracingCustomTags() {
+  if (!hasCachedRoute() || !cached_route_.value()->tracingConfig()) {
+    tracing_custom_tags_.clear();
+    return;
+  }
+  const std::vector<Tracing::CustomTag>& route_tags =
+      cached_route_.value()->tracingConfig()->getCustomTags();
+  if (route_tags.empty()) {
+    tracing_custom_tags_.clear();
+    return;
+  }
+  auto cmp = [](const Tracing::CustomTag& a, const Tracing::CustomTag& b) { return a.tag < b.tag; };
```

Comment:
instead of passing a lambda you i believe you can just define `operator<` on CustomTag

Reply:
use a map